src/main/resources/test_files/runtime/FunctionArrayDescendant.iq

```diff
@@ -0,0 +1,4 @@
+(:JIQS: ShouldRun; Output="( [ 0, x, { "a" : [ 1, { "b" : 2 }, [ 2.5 ] ], "o" : { "c" : 3 } } ], [ 1, { "b" : 2 }, [ 2.5 ] ], [ 2.5 ])" :)
+let $o := ([0, "x", { "a" : [1, {"b" : 2}, [2.5]], "o" : {"c" : 3} }]) return descendant-arrays($o)
```

Comment:
We need to test corner cases (top-level object, flat array, empty array, empty object)